src/func.d

```diff
@@ -3029,6 +3008,46 @@ extern (C++) class FuncDeclaration : Declaration
         return true; // functions can be overloaded
     }
 
+    /**********************************
+     * Decide if attributes for this function can be inferred from examining
+     * the function body.
+     * Returns:
+     *  true if can
+     */
+    private final bool canInferAttributes(Scope* sc)
+    {
+        TemplateInstance ti;
+        return (fbody && !isVirtualMethod() &&
+                        /********** this is for backwards compatibility for the moment ********/
```

Comment:
Below, and the beginning/end of the comments line up with what it refers to. In the next PR, I plan to straighten out the logic here. But in order to make this refactor easy to review, I stayed with it being a mere cut/paste of the original code.